File: MMORPGServer/Proto/CreateLuaProto/WorldMap_OtherRoleResurgenceProto.lua
--服务器广播角色复活消息
WorldMap_OtherRoleResurgenceProto = { ProtoCode = 13016, RoleId = 0 }

--这句是重定义元表的索引，就是说有了这句，这个才是一个类
WorldMap_OtherRoleResurgenceProto.__index = WorldMap_OtherRoleResurgenceProto;

function WorldMap_OtherRoleResurgenceProto.New()
    local self = { }; --初始化self
    setmetatable(self, WorldMap_OtherRoleResurgenceProto); --将self的元表设定为Class
    return self;
end


--发送协议
function WorldMap_OtherRoleResurgenceProto.SendProto(proto)

    local ms = CS.LuaHelper.Instance:CreateMemoryStream();
    ms:WriteUShort(proto.ProtoCode);

    ms:WriteInt(proto.RoleId);

    CS.LuaHelper.Instance:SendProto(ms:ToArray());
    ms:Dispose();
end


--解析协议
function WorldMap_OtherRoleResurgenceProto.GetProto(buffer)

    local proto = WorldMap_OtherRoleResurgenceProto.New(); --实例化一个协议对象
    local ms = CS.LuaHelper.Instance:CreateMemoryStream(buffer);

    proto.RoleId = ms:ReadInt();

    ms:Dispose();
    return proto;
end

File: mmoTest/Assets/Script/Data/Proto/WorldMap/WorldMap_OtherRoleResurgenceProto.cs
//===================================================
//作    者：边涯  http://www.u3dol.com  QQ群：87481002
//创建时间：2018-02-25 22:40:38
//备    注：
//===================================================
using System.Collections;
using System.Collections.Generic;
using System;

/// <summary>
/// 服务器广播角色复活消息
/// </summary>
public struct WorldMap_OtherRoleResurgenceProto : IProto
{
    public ushort ProtoCode { get { return 13016; } }

    public int RoleId; //角色编号

    public byte[] ToArray()
    {
        using (MMO_MemoryStream ms = new MMO_MemoryStream())
        {
            ms.WriteUShort(ProtoCode);
            ms.WriteInt(RoleId);
            return ms.ToArray();
        }
    }

    public static WorldMap_OtherRoleResurgenceProto GetProto(byte[] buffer)
    {
        WorldMap_OtherRoleResurgenceProto proto = new WorldMap_OtherRoleResurgenceProto();
        using (MMO_MemoryStream ms = new MMO_MemoryStream(buffer))
        {
            proto.RoleId = ms.ReadInt();
        }
        return proto;
    }
}